src/ImageReconstruction.jl

```diff
@@ -60,41 +50,58 @@ function _ramp_spatial(N::Int, τ)
     h
 end
 
-_zero_pad(p::AbstractVector, N::Int) = vcat(p, zeros(N))
-
 """
-Inverse radon transform using a ramp filter and a pixel-driven algorithm.
+	iradon(P::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
+
+Inverse radon transform of a sinogram `P` with view angles `θ` in radians and
+detector sampling `t` producing an image on a 128x128 matrix.
 """
-function iradon(sinogram::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
+function iradon(P::AbstractMatrix, θ::AbstractRange, t::AbstractRange)
     pixels = 128
 
     N = length(t)
     K = length(θ)
     Npad = nextpow(2, 2 * N - 1)
     τ = step(t)
-    ramp = fft(_zero_pad(_ramp_spatial(N, τ), Npad - N))
-    i = div(N, 2) + 1
+    ramp = fft(_ramp_spatial(N, τ, Npad))
+    i = N ÷ 2 + 1
     j = i + N - 1
 
-    image = zeros(eltype(sinogram), pixels, pixels)
-    Q = Vector{eltype(sinogram)}(undef, N)
-    for (k, θₖ) in enumerate(θ)
+    T = eltype(P)
+    image = [zeros(T, pixels, pixels) for _ = 1:nthreads()]
+    P′ = [Vector{Complex{T}}(undef, Npad) for _ = 1:nthreads()]
+    Q = [Vector{T}(undef, N) for _ = 1:nthreads()]
+
+    l = SpinLock()
+    @inbounds @threads for (k, θₖ) in collect(enumerate(θ))
+        id = threadid()
+
         # filter projection
-        Q[:] .= τ .* real.(ifft(fft(_zero_pad(sinogram[:, k], Npad - N)) .* ramp)[i:j])
+        P′[id][1:N] .= P[:, k]
```

Comment:
Assign `P′[id]` to a local variable? If nothing else it just seems prettier than repeated indexing. You might even have
```julia
Pid, Qid, imageid = P′[id], Q[id], image[id]
```
at the top of the loop body and then the rest of the code mostly doesn't even care that you're splitting this out into threads.